Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3142,18 +2045,19 @@ private void setActivityLifeCycleObserver(Application application) {
 
         @Override
         public void onActivityCreated(Activity activity, Bundle bundle) {
+            intentState_ = INTENT_STATE.PENDING;
             if (BranchViewHandler.getInstance().isInstallOrOpenBranchViewPending(activity.getApplicationContext())) {
                 BranchViewHandler.getInstance().showPendingBranchView(activity);
             }
         }
 
         @Override
         public void onActivityStarted(Activity activity) {
+            intentState_ = INTENT_STATE.PENDING;
```

Comment:
The latency with this solution raelly depends on where the ini session is
called. Assuming it is called in "onstart" the latency really depends on
the code added to the onStart
On Jul 21, 2016 11:01 PM, "Alex Austin" <notifications@github.com> wrote:

> In Branch-SDK/src/io/branch/referral/Branch.java
> <https://github.com/BranchMetrics/android-branch-deep-linking/pull/271#discussion_r71830252>
> :
>
> >              if (BranchViewHandler.getInstance().isInstallOrOpenBranchViewPending(activity.getApplicationContext())) {
> >                  BranchViewHandler.getInstance().showPendingBranchView(activity);
> >              }
> >          }
> >
> >          @Override
> >          public void onActivityStarted(Activity activity) {
> > +            intentState_ = INTENT_STATE.PENDING;
>
> And if so, how much additional latency do you anticipate this introducing?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BranchMetrics/android-branch-deep-linking/pull/271/files/5a7e8a3a10d3a3a67061a05f7a23c9124671d6ed#r71830252>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHTA9CykUyEsqeI343wHKumdCxRubuvRks5qYFyqgaJpZM4JSWfm>
> .
>